server/log/router.ts

```diff
@@ -14,8 +21,51 @@ export const LogModule: UIServerModule = {
     const routerForModule = express.Router();
 
     // implementation to follow
-    routerForModule.get('*', authFn, (req, res) => res.sendStatus(200));
+    routerForModule.get('*', authFn, (req, res) => {
+      const { isWs } = req as strimziUIRequestType;
+      const { ws } = res as strimziUIResponseType;
+      if (isWs) {
+        ws.on('message', handleMessage(logger));
+        ws.on('close', handleClose(logger));
+      } else {
+        // Return 426 Upgrade Required if this isn't a websocket request
+        res.sendStatus(426);
+      }
+    });
 
     return exit({ mountPoint: '/log', routerForModule });
   },
 };
+
+const handleMessage: (logger: Logger) => (data: Data) => void = (logger) => (
+  data
+) => {
+  if (typeof data === 'string') {
+    try {
+      JSON.parse(data).forEach((clientLogEvent: ClientLoggingEvent) => {
+        if (clientLogEvent.clientLevel) {
+          logger[clientLogEvent.clientLevel](
+            clientLogEvent,
+            clientLogEvent.msg
+          );
+        } else {
+          logger.debug(clientLogEvent, clientLogEvent.msg);
+        }
+      });
+    } catch (error) {
+      // Ignore any data that cannot be parsed
+      logger.trace(
+        { error },
+        `handleMessage failed: ${error.message}, ${data}`
+      );
+    }
+  } else {
+    // Ignore any non-string data
+    logger.trace(`handleMessage ignoring data of type ${typeof data}`);
+  }
+};
+
+const handleClose: (
+  logger: Logger
+) => (code: number, reason: string) => void = (logger) => (code, reason) =>
+  logger.debug(`WebSocket listener closed. (${code}) ${reason}`);
```

Comment:
Can we log out a request ID / client ID?

Reply:
The close event only receives the code and reason string: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ws/index.d.ts#L86 so we would need to encode the client ID in the reason string. I will look into this when I hook into the context, as we may want to explicitly close the logging websocket on client unload.